enhancements/external-secrets-operator/external-secrets-component-config.md

```diff
@@ -0,0 +1,321 @@
+- Provide a declarative API for adding custom annotations globally to all component Deployments and Pod templates via `controllerConfig.annotations`.
+- Provide a declarative API for specifying custom environment variables for each component via `componentConfig.overrideEnv`.
+- Support all four operand components: Controller, Webhook, CertController, and BitwardenSDKServer.
+
+### Non-Goals
+
+- Exhaustive validation of individual argument values. Users should consult upstream documentation. Only basic structural validation (non-empty strings, list length limits) will be performed. Invalid arguments will result in container runtime failures (CrashLoopBackOff) rather than API rejection.
+- Resource limits, replica counts, or other deployment-level settings except for the RevisionHistoryLimit which is specifically introduced by this proposal.
+
+## Proposal
+
+Extend the ControllerConfig API with:
+1. A new `annotations` field for adding custom annotations globally to Deployments and Pod templates.
+2. A new `componentConfig` field for per-component deployment lifecycle overrides.
+
+### Workflow Description
+
+**For Global Annotations:**
+
+1. **User Configuration:** Administrator updates the `ExternalSecretsConfig` CR with the `controllerConfig.annotations` field containing custom key-value pairs.
+2. **Validation:** The operator validates that annotation keys and values conform to Kubernetes annotation constraints.
+3. **Reconciliation:** The operator merges user-specified annotations with any default annotations. User annotations take precedence in case of conflicts. Annotations are applied to both the Deployment metadata and Pod template metadata for all components.
+4. **Rollout:** Kubernetes detects the annotation changes and performs updates as needed.
+
+**For Component Configuration:**
+
+1. **User Configuration:** Administrator updates the `ExternalSecretsConfig` CR, utilizing the new `componentConfig` list to specify configuration entries for a component (Controller, Webhook, etc.). This includes deployment-level overrides via `overrideArgs` and custom environment variables via `overrideEnv`.
+2. **Validation:** It verifies the `componentName` against the allowed enum values and enforces uniqueness across the list. It strictly validates the `OverrideArgs` field using the provided `XValidation` rule, ensuring every entry uses the specified format. For `overrideEnv`, it validates that environment variable names and values conform to Kubernetes conventions.
+3. **Reconciliation:** It parses the `OverrideArgs` field to identify and extract the deployment override key and its corresponding value. It updates the component's underlying Kubernetes Deployment resource by setting the parsed override value in the appropriate `.spec` field. For `overrideEnv`, the operator merges user-specified environment variables with default variables, with user values taking precedence in case of conflicts.
```

Comment:
```suggestion
3. **Reconciliation:** It parses the `overrideDeploymentConfig` field to identify and extract the deployment override key and its corresponding value. It updates the component's underlying Kubernetes Deployment resource by setting the parsed override value in the appropriate `.spec` field. For `overrideEnv`, the operator merges user-specified environment variables with default variables, with user values taking precedence in case of conflicts.
```